codegenerator/cli/templates/static/codegen/src/EventFetching.res

```diff
@@ -93,26 +140,37 @@ let getNextPage = (
     [queryTimoutPromise, logsPromise]
     ->Promise.race
     ->Promise.catch(err => {
-      // Might fail with message "query exceeds max results 20000, retry with the range 6000438-6000934"
-      // Ideally to handle it and respect
-
-      logger->Logging.childWarn({
-        "msg": "Error getting events. Will retry after backoff time",
-        "backOffMilliseconds": sc.backoffMillis,
-        "suggestedBlockInterval": suggestedBlockInterval,
-        "err": err,
-      })
+      switch getSuggestedBlockIntervalFromExn(err) {
```

Comment:
It might be worth first checking if it was timeout from the race but probably not going to be a big deal for performance either way.